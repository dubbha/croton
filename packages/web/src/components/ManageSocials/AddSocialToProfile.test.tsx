import React from 'react';
import { render } from '@testing-library/react';

describe('components/AddSocialToProfile', () => {
  const getGenericSocialAPI = () => ({
    onResponseAction,
    onErrorAction,
    errorMessage,
    buttonText,
  }) => (
    <div data-testid="genericSocialApi">
      <div>{onResponseAction}</div>
      <div>{onErrorAction}</div>
      <div>{errorMessage}</div>
      <div>{buttonText}</div>
    </div>
  );

  const setUpMocks = ({ socialProfile }) => {
    jest.doMock('react-redux', () => ({
      useSelector: () => ({ socialProfile }),
    }));

    jest.doMock('./FbAuth', () => ({
      FbAuth: getGenericSocialAPI(),
    }));

    jest.doMock('./GoogleAuth', () => ({
      GoogleAuth: getGenericSocialAPI(),
    }));

    const { AddSocialToProfile } = require('./AddSocialToProfile');

    return render(<AddSocialToProfile />);
  };

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render add social profile buttons if no social profile has been added yet', () => {
    jest.isolateModules(() => {
      const { container } = setUpMocks({ socialProfile: null });

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should not render add social profile buttons if a social profile has been added', () => {
    jest.isolateModules(() => {
      const { container } = setUpMocks({
        socialProfile: { some: 'profile' },
      });

      expect(container.firstChild).toEqual(null);
    });
  });
});
