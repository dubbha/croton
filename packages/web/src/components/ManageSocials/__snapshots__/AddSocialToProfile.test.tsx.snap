// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AddSocialToProfile should render add social profile buttons if no social profile has been added yet 1`] = `
<div
  class="authButtonContainer"
>
  <div
    data-testid="genericSocialApi"
  >
    <div>
      ADD_FACEBOOK
    </div>
    <div>
      ADD_FACEBOOK_ERROR
    </div>
    <div>
      Sorry, something went wrong with adding your Facebook account
    </div>
    <div>
      Add Facebook to profile
    </div>
  </div>
</div>
`;
